hub/core/storage/cachable.py

```diff
@@ -43,7 +43,8 @@ def copy(self):
     @classmethod
     def frombuffer(cls, buffer: bytes):
         instance = cls()
-        instance.__setstate__(json.loads(buffer))
+        if len(buffer) > 0:
+            instance.__setstate__(json.loads(buffer))
```

Comment:
I get that this will fix the error in which we get an empty buffer but wouldn't this bubble up elsewhere as we'll create an empty Cachable object now?
I think it's more important to figure out why the buffer is empty in the first place.

Reply:
yes likely it is related to passing dataset meta info cache through pickling to forked processes